File: Data Structures & Algorithms/duplicate-integer/submission-0.js
class Solution {
    /**
     * @param {number[]} nums
     * @return {boolean}
     */
    hasDuplicate(nums) {
        let map = new Set()
        for(let num of nums){
            if(map.has(num)){
                return true
            }
            map.add(num)
        }
        return false
    }
}

File: Data Structures & Algorithms/missing-number/submission-0.cpp
class Solution {
public:
    int missingNumber(vector<int>& nums) {
        int n = nums.size();
        int xor1 = 0;
        int xor2 = 0;
        for(int i = 0; i <= n; i++){
            xor1 ^= i;
        }    
        for(const int& num:nums){
            xor2 ^= num;
        }    

        return xor1^xor2;
    }
};
